hw3/01-characters.js

```diff
@@ -1,4 +1,63 @@
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const url = "https://thronesapi.com/api/v2/Characters/rgdr";
 
-fetch(url);
+const charContainer = document.querySelector("#characters-container");
+charContainer.classList.add("card-deck");
+const errorMessage = document.createElement("li");
+const printCharacters = async function () {
+  try {
+    const response = await fetch(url);
+    const jsonData = await response.json();
+    errorMessage.remove();
+    jsonData.forEach((element) => {
+      const { fullName, title, family, imageUrl } = element;
+      //card
+      const charDiv = document.createElement("li");
+      charDiv.classList.add("card");
+
+      //card image
+      const charImage = document.createElement("img");
+      charImage.classList.add("card-img");
+      charImage.src = imageUrl;
+
+      if (family.includes("House")) {
+        charImage.alt = `${fullName} of ${family}`;
+      } else {
+        charImage.alt =
+          family.includes("Free") ||
+          family.includes("Unk") ||
+          family.includes("None")
+            ? `${fullName} - ${title}`
+            : `${fullName} of House ${family}`;
+      }
+      //card-body div
+      const charNameDiv = document.createElement("div");
+      charNameDiv.classList.add("card-body");
+
+      //card-body-text fullName
+      const charName = document.createElement("span");
+      charName.classList.add("card-title");
+      charName.innerHTML = fullName;
```

Comment:
It would make more sense to use a figcaption, or even a heading for the character name. It would be more semantic than just using a span tag.

Also, why is innerHTML being used, instead of textContent?